actors/builtin/miner/policy.go

```diff
@@ -90,12 +91,57 @@ var SealedCIDPrefix = cid.Prefix{
 	MhLength: 32,
 }
 
-// List of proof types which may be used when creating a new miner actor.
+// List of proof types which may be used when creating a new miner actor or pre-committing a new sector.
 // This is mutable to allow configuration of testing and development networks.
-var SupportedProofTypes = map[abi.RegisteredSealProof]struct{}{
+var PreCommitSealProofTypesV0 = map[abi.RegisteredSealProof]struct{}{
 	abi.RegisteredSealProof_StackedDrg32GiBV1: {},
 	abi.RegisteredSealProof_StackedDrg64GiBV1: {},
 }
+var PreCommitSealProofTypesV7 = map[abi.RegisteredSealProof]struct{}{
+	abi.RegisteredSealProof_StackedDrg32GiBV1:   {},
+	abi.RegisteredSealProof_StackedDrg64GiBV1:   {},
+	abi.RegisteredSealProof_StackedDrg32GiBV1_1: {},
+	abi.RegisteredSealProof_StackedDrg64GiBV1_1: {},
+}
+
+// From network version 8, sectors sealed with the V1 seal proof types cannot be committed.
+var PreCommitSealProofTypesV8 = map[abi.RegisteredSealProof]struct{}{
+	abi.RegisteredSealProof_StackedDrg32GiBV1_1: {},
+	abi.RegisteredSealProof_StackedDrg64GiBV1_1: {},
+}
+
+// Checks whether a seal proof type is supported for new miners and sectors.
+func CanPreCommitSealProof(s abi.RegisteredSealProof, nv network.Version) bool {
+	_, ok := PreCommitSealProofTypesV0[s]
+	if nv >= network.Version7 {
```

Comment:
Is there a reason not to reverse the order and use else ifs/elses? The extra map hits aren't going to cost much, but it looks a little sloppy to my eye.

Reply:
My motivation for this pattern is only to maintain the pattern that every network version check takes the form `nv >= network.VersionX`, and to also maintain a pattern that the cases for different versions are presented in their historical order. With only one way to do it, there's less thought and chance of error. 

However, we could re-order these and maintain the first property (but not the second) and I think it would be fine.